File: minimumBribe.py


def minimumBribes(queue):
    # Write your code here
    bribe = 0
    for i in range(len(queue)):
        if queue[i] - (i + 1) > 2:
            print("Too chaotic")
            return
        for j in range(max(0, queue[i] - 2), i):
            if queue[j] > queue[i]:
                bribe += 1
    return bribe

print(minimumBribes([1,2,3,4,5,7,6,8]))

File: MinimumBribe.java
public class MinimumBribe {
    public static void main(String[] args) {
        int[] q = {2, 1, 5, 3, 4};
        System.out.println(minimumBribe(q));
    }

    public static int minimumBribe(int[] q) {
        int bribes = 0;
        for (int i = 0; i < q.length; i++) {
            if (q[i] - (i + 1) > 2) {
                System.out.println("Too chaotic");
                return 0;
            }
            for (int j = Math.max(0, q[i] - 2); j < i; j++) {
                if (q[j] > q[i]) {
                    bribes++;
                }
            }
        }
        System.out.println(bribes);
        return bribes;
    }
}
